mockery/outputter_test.go

```diff
@@ -0,0 +1,36 @@
+package mockery
+
+import (
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+)
+
+func TestFilenameBare(t *testing.T) {
+	out := FileOutputStreamProvider{InPackage: false, TestOnly: false}
+	assert.Equal(t, "name.go", out.filename("name"))
+}
+
+func TestFilenameMockOnly(t *testing.T) {
+	out := FileOutputStreamProvider{InPackage: true, TestOnly: false}
+	assert.Equal(t, "mock_name.go", out.filename("name"))
+}
+
+func TestFilenameMockTest(t *testing.T) {
+	out := FileOutputStreamProvider{InPackage: true, TestOnly: true}
+	assert.Equal(t, "mock_name_test.go", out.filename("name"))
+}
+
+func TestFilenameTest(t *testing.T) {
+	out := FileOutputStreamProvider{InPackage: false, TestOnly: true}
+	assert.Equal(t, "name_test.go", out.filename("name"))
+}
+
+func TestUnderscoreCaseName(t *testing.T) {
+	assert.Equal(t, "notify_event", (&FileOutputStreamProvider{}).underscoreCaseName("NotifyEvent"))
```

Comment:
I know this is ugly, basically testing a private method. I think it'll be at a point soon where the function will be testable without the struct again. Maybe with a smaller interface.